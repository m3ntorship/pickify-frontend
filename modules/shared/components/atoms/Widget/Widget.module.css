.widget-box {
  @apply bg-grey-bg rounded-md shadow-soft py-4 px-4;
}

.widget-box p {
  @apply font-normal text-grey text-sm;
}
