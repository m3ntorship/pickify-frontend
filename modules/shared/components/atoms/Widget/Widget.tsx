import React from 'react';
import type { FC, ReactElement } from 'react';
import styles from './Widget.module.css';

const Widget: FC = ({ children }): ReactElement => {
  return (
    <div className={styles['widget-box']}>
      <p>{children}</p>
    </div>
  );
};

export default Widget;
