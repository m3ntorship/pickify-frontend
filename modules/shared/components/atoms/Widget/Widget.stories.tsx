import React from 'react';
import type { Story, Meta } from '@storybook/react';
import Widget from './Widget';

export default {
  title: 'Atoms/Widget',
  component: Widget,
} as Meta;

const Template: Story = (args) => (
  <Widget {...args}>
    Lorem Ipsum is simply dummy text of the printing and typesetting industry.
  </Widget>
);

export const widget = Template.bind({});
