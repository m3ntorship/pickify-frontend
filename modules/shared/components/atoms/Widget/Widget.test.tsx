import React from 'react';
import renderer from 'react-test-renderer';
import Widget from './Widget';

describe('Widget', () => {
  it('should render Widget successfully', () => {
    const tree = renderer
      .create(
        <Widget>
          Lorem Ipsum is simply dummy text of the printing and typesetting
          industry.
        </Widget>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
