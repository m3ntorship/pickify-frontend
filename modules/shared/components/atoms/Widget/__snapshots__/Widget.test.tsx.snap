// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget should render Widget successfully 1`] = `
<div
  className="widget-box"
>
  <p>
    Lorem Ipsum is simply dummy text of the printing and typesetting industry.
  </p>
</div>
`;
